graylog2-server/src/main/java/org/graylog2/rest/RestTools.java

```diff
@@ -112,7 +111,13 @@ public static URI buildExternalUri(@NotNull MultivaluedMap<String, String> httpH
                     .findFirst();
         }
 
-        return externalUri.orElse(defaultUri);
+        final URI uri = externalUri.orElse(defaultUri);
+
+        // Make sure we return an URI object with a trailing slash
+        if (!uri.toString().endsWith("/")) {
```

Comment:
What happens if `externalUri` already has *multiple* trailing slashes, i. e. `http://example.org///´?

Reply:
In this case it doesn't add another trailing slash. But multiple trailing slashes are not cleaned up. That didn't change.